libs/common-system/src/main/java/org/codice/ddf/configuration/AbsolutePathResolver.java

```diff
@@ -53,4 +58,58 @@ public String getPath() {
     String absolutePathStr = absolutePath.toAbsolutePath().toString();
     return trailingSeparator ? absolutePathStr + File.separator : absolutePathStr;
   }
+
+  /**
+   * This function returns the absolute path of its given path. It checks the given {@code
+   * systemProperties} in order and returns the first one that isn't null (or empty depending on the
+   * value of {@code allowEmpty})
+   *
+   * @param systemProperties list of strings of system properties to try and append to the path
+   * @param allowEmpty if a system property is an empty string (i.e "") this will be valid and
+   *     appended to the path
+   * @param allowToAbs will call the {@code path.toAbsolutePath} at the end of the function
+   * @return the absolute path
+   */
+  public String getPath(List<String> systemProperties, boolean allowEmpty, boolean allowToAbs) {
```

Comment:
❓ Not sure why this method is needed. I seriously think this is overkill for what we were trying to accomplish. We just needed to extend support for "karaf.home" in `AbsolutePathResolver` without adding too much additional functionality.